### PR TITLE
debug an exception

### PR DESCRIPTION
**Pull request by LowinLi**
if dst_path is not a file, it will raise Exception in the function src_path.samefile:
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/lilongwei/notebook/onnx_diffusion/vae_decoder/model.onnx'
```

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
cc @anton-l

**Comment by patrickvonplaten** [comment]
cc @anton-l could you take a look here?

File: src/diffusers/onnx_utils.py [hunk 1]
```diff
@@ -79,7 +79,9 @@ def _save_pretrained(self, save_directory: Union[str, Path], file_name: Optional
 
         src_path = self.model_save_dir.joinpath(self.latest_model_name)
         dst_path = Path(save_directory).joinpath(model_file_name)
-        if not src_path.samefile(dst_path):
+        if dst_path.is_file() and not src_path.samefile(dst_path):
+            shutil.copyfile(src_path, dst_path)
+        elif not dst_path.is_file():
             shutil.copyfile(src_path, dst_path)
```
Comment (anton-l) [inline src/diffusers/onnx_utils.py:85]: Looks like we can just do
```suggestion
        shutil.copyfile(src_path, dst_path)
```

**Review by anton-l — APPROVED** [review]
Thank you for catching the bug @LowinLi!
Just a quick update, since the if-else branches turned out to be the same:

File: src/diffusers/onnx_utils.py (hunk 1 above)
Comment (LowinLi) [reply src/diffusers/onnx_utils.py]: Sure, it seems great. Thanks

**Pull request merged**